### PR TITLE
fix: make commit sha from main

### LINKED ISSUE #8295: jx version shows commit sha which is not in main branch
```
(base) ➜  ~ jx version
version: 3.3.0
shaCommit: 7e8722c04c1d6f85173631ac671d76178736d507
buildDate: Fri Jul  8 16:24:34 UTC 2022
goVersion: 1.17.9
branch: main
gitTreeState: clean
```

commit sha `7e8722c04c1d6f85173631ac671d76178736d507` is not in the main branch.
image

it is created in the pipeline. So, it's incorrect that the branch says main. 
It should be fixed to show a commit in the main branch.

### PR DESCRIPTION
**Pull request by osamamagdy**
Signed-off-by: osamamagdy <osamamagdy174@gmail.com>

fixes #8295

### CODE REVIEW

File: .github/workflows/jenkins-x/upload-binaries.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@ git config --global user.email "jenkins-x@googlegroups.com"
 
 export BRANCH=$(git rev-parse --abbrev-ref HEAD)
 export BUILDDATE=$(date)
-export REV=$(git rev-parse HEAD)
+export REV=$(git rev-parse --short origin/main 2> /dev/null || echo 'unknown')
```
Comment (ankitm123) [inline .github/workflows/jenkins-x/upload-binaries.sh:15]: If HEAD resolved to main in L13, I dont think this fixes it. The commit actually is not in main branch, I posted the screenshot in the issue ...

Comment (osamamagdy) [reply .github/workflows/jenkins-x/upload-binaries.sh]: Ok, applying another way to not create the empty commit

File: .github/workflows/jenkins-x/changelog.sh [hunk 2]
```diff
@@ -13,9 +13,7 @@ echo "setting git user"
 git config --global user.name jenkins-x-bot-test
 git config --global user.email "jenkins-x@googlegroups.com"
 
-git add * || true
-git commit -a -m "chore: release $VERSION" --allow-empty
-git tag -fa v$VERSION -m "Release version $VERSION"
+git tag -fa v$VERSION -m "Release version $VERSION" HEAD
```
Comment (ankitm123) [inline .github/workflows/jenkins-x/changelog.sh:16]: coventional commit message format always.

Comment (osamamagdy) [reply .github/workflows/jenkins-x/changelog.sh:16]: start it with `chore:`  ?

File: .github/workflows/jenkins-x/changelog.sh [hunk 3]
```diff
@@ -13,9 +13,7 @@ echo "setting git user"
 git config --global user.name jenkins-x-bot-test
 git config --global user.email "jenkins-x@googlegroups.com"
 
-git add * || true
-git commit -a -m "chore: release $VERSION" --allow-empty
-git tag -fa v$VERSION -m "Release version $VERSION"
+git tag -fa v$VERSION -m "chore: release version $VERSION" HEAD
```
Comment (ankitm123) [inline .github/workflows/jenkins-x/changelog.sh:16]: Why head at the end?

Comment (osamamagdy) [reply .github/workflows/jenkins-x/changelog.sh:16]: I was trying to make it obvious that you tag the last commit and if you want to go back you can do `HEAD~1`

Comment (ankitm123) [reply .github/workflows/jenkins-x/changelog.sh:16]: It's the default, we can skip imo.

**Pull request merged**